### PR TITLE
Topic/tcp code cleanup

### PR DESCRIPTION
**Pull request by alex-mikheev**
Highly experimental changes to lwip and vma code that improve tcp latency

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/200/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/214/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
@alex-mikheev I think that some work is done in https://github.com/Mellanox/libvma/commit/2c2c665a
probably related https://github.com/alex-mikheev/libvma/commit/8d10c9b4b. Right?

**Comment by alex-mikheev** [comment]
right. I tried to remove all logging from the critical path

File: src/vma/dev/ring_simple.cpp [hunk 1]
```diff
@@ -687,6 +689,60 @@ rfs *ring_simple::find_tcp_flow(mem_buf_desc_t *p_rx_wc_buf_desc, struct iphdr*
...
+		p_rx_wc_buf_desc->sz_data -= (sz_data - ip_tot_len);
+	}
+
+	p_rx_wc_buf_desc->path.rx.vma_polled = true;
+    // AM_TODO: virtual call
+	p_rfs->rx_dispatch_packet(p_rx_wc_buf_desc, &m_vma_comp_arr[m_vma_curr_comp_index]);
+	p_rx_wc_buf_desc->path.rx.vma_polled = false;
+}
+#endif
+
```
Comment (igor-ivanov) [inline src/vma/dev/ring_simple.cpp]: Do we remove statistics in this code?

File: src/vma/lwip/opt.h [hunk 2]
```diff
@@ -1411,7 +1411,7 @@
  * use custom congestion control algorithms
  */
 #ifndef TCP_CC_ALGO_MOD
-#define TCP_CC_ALGO_MOD 1
+#define TCP_CC_ALGO_MOD 0
 #endif
```
Comment (igor-ivanov) [inline src/vma/lwip/opt.h]: What improvement should be seen?

File: src/vma/lwip/opt.h [hunk 3]
```diff
@@ -1544,7 +1544,7 @@
  * LWIP_TCP_TIMESTAMPS==1: support the TCP timestamp option.
  */
 #ifndef LWIP_TCP_TIMESTAMPS
-#define LWIP_TCP_TIMESTAMPS             1
+#define LWIP_TCP_TIMESTAMPS             0
 #endif
```
Comment (igor-ivanov) [inline src/vma/lwip/opt.h]: Do you see any limitation after this change?

**Comment by OphirMunk** [comment]
@alex-mikheev great execution! 
We will split the commits into individual PRs and close this one.

**Pull request closed without merging**